html/browsers/origin/origin-of-data-document.html

```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <title>Origin of document produced from a 'data:' URL</title>
+    <link rel="help" href="https://html.spec.whatwg.org/multipage/browsers.html#origin">
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <script>
+      async_test(function (t) {
+        window.addEventListener("message", t.step_func_done(function (e) {
+          assert_equals(e.origin, "null", "Messages sent from a 'data:' URL should have an opaque origin (which serializes to 'null').");
+          assert_throws("SecurityError", function () {
+            var couldAccessCrossOriginProperty = e.source.location.href;
+          }, "The 'data:' frame should be cross-origin.")
+        }));
+
+        var i = document.createElement('iframe');
+        i.src = "data:text/html,<script>" +
+                "  window.top.postMessage('Hello!', '*');" +
```

Comment:
Use `parent` instead of `top`

> @zcorpan are there test runners that run tests in a way such that window.top in a test will point to the runner rather than the test?
@zcorpan or is it fine to use window.top and expect the test
@Ms2ger I'd say use it and anyone who has problems with it can submit a pr
@jgraham zcorpan: There are, but I'm pretty sure there are already tests broken in that environment
@jgraham e.g. some in-browser harnesses written for tv and similar environments that don't support window.open